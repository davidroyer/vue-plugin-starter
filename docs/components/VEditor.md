# v-editor 

An Amazing Editor Component 

- **author** - David Royer 
- **license** - MIT 
- **VEditor** 

## slots 

- `default`  

- `label` Use this slot to set the checkbox label 

## props 

- `msg` ***String*** (*optional*) 

   Test prop for unit tests 

- `model` ***Array*** (*optional*) 

   The checkbox model 

- `disabled` ***Boolean*** (*optional*) 

   Initial checkbox state 

- `enabled` ***Boolean*** (*optional*) `default: true` 

   Initial checkbox value 

- `label` ***String*** (*optional*) `default: 'Unamed checkbox'` 

   The checkbox label 

- `object` ***Object*** (*optional*) `default: null` 

- `bool-false` ***Boolean*** (*optional*) `default: false` 

## data 

- `initialValue` 

   The initial component value. Used to detect changes and restore the initial value. 

**initial value:** `'The initial value!'` 

- `currentValue` 

**initial value:** `'And the current value!'` 

## computed properties 

- `id` 

   The component identifier. Generated using the `initialValue` data. 

   **dependencies:** `initialValue` 

- `changed` 

   **dependencies:** `currentValue`, `initialValue` 

- `withNoDependencies` 

## events 

- `loaded` 

   Emitted when the component has been loaded 

- `enabled` 

   Emitted the event `enabled` when loaded Multilign 

## methods 

- `check()` 

   Check if the input is checked 

- `prop()` 

- `enable(value)` 

   Enable the checkbox 

