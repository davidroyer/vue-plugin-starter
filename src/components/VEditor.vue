<template>
<label>
    <h1>{{ msg }}</h1>
    <input :id='id' :disabled="disabled" type="text" v-model="checkbox">
    <slot></slot>
    <!-- Use this slot to set the checkbox label -->
    <slot name="label"></slot>
  </label>
</template>

<script>
/**
 * An Amazing Editor Component
 *
 * @author David Royer
 * @license MIT
 * @VEditor
 */
export default {
  name: "VEditor",
  props: {
    /**
     * Test prop for unit tests
     */
    msg: String,

    /**
     * The checkbox model
     */
    model: {
      type: Array
      // required: true
    },

    /**
     * Initial checkbox state
     */
    disabled: Boolean,

    /**
     * Initial checkbox value
     */
    enabled: {
      type: Boolean,
      default: true
    },

    /**
     * The checkbox label
     */
    label: {
      type: String,
      default: "Unamed checkbox"
    },

    object: {
      type: Object,
      default: null
    },

    boolFalse: {
      type: Boolean,
      default: false
    }
  },

  created() {
    /**
     * Emitted when the component has been loaded
     */
    this.$emit("loaded");
  },

  data() {
    return {
      /**
       * The initial component value.
       * Used to detect changes and restore the initial value.
       */
      initialValue: "The initial value!",
      currentValue: "And the current value!",
      test1: "fix"
    };
  },

  computed: {
    /**
     * The component identifier.
     * Generated using the `initialValue` data.
     */
    id() {
      return `checkbox-${this.initialValue}`;
    },

    changed() {
      return this.currentValue !== this.initialValue;
    },

    withNoDependencies() {
      return "yeah";
    }
  },

  methods: {
    /**
     * Check if the input is checked
     */
    check() {
      console.log("check");
    },

    prop: () => {
      console.log("prop");
    },

    /*
     * This should be ignored
     * @private
     */
    ignored() {},

    // Another ignored method
    // @private
    anotherIgnored() {},

    // Enable the checkbox
    enable(value) {
      const x = true;
      /**
       * Emitted the event `enabled` when loaded
       * Multilign
       */
      this.$emit("enabled", x);
    }
  }
};
</script>

<style lang="css" scoped>
</style>
