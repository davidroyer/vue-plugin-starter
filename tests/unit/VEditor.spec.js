import { mount } from "@vue/test-utils";
import VEditor from "@/components/VEditor.vue";

describe("VEditor.vue", () => {
  it("renders component", () => {
    const wrapper = mount(VEditor);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders props.msg when passed", () => {
    const msg = "new message";
    const wrapper = mount(VEditor, {
      propsData: { msg }
    });
    expect(wrapper.text()).toMatch(msg);
  });
});
