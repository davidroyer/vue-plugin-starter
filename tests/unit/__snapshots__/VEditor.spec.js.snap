// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VEditor.vue renders component 1`] = `
<label>
  <h1>
    
  </h1>
   
  <input
    id="checkbox-The initial value!"
    type="text"
  />
    
</label>
`;
